File: client/src/app/reviews/review.ts
// export class Review {

//     #id: number

//     #title: string

//     #description: string

//     #rating: number

//     constructor(_id:number,_title:string,_description:string,_rating:number){
//         this.#id=_id;
//         this.#title=_title;
//         this.#description=_description;
//         this.#rating=_rating
//     }
// }

export interface Review {

    id: number

    title: string

    description: string

    rating: number

}

File: server/src/main/java/com/app/server/product/domain/entity/Review.java
package com.app.server.product.domain.entity;

import com.app.server.customer.domain.entity.Customer;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "review")
public class Review {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String title;

  private String description;

  private int rating;

  @ManyToOne
  @JoinColumn(name = "id_customer", nullable = false)
  private Customer customer;

  @ManyToOne
  @JoinColumn(name = "id_product", nullable = false)
  private Product product;
}
